### ISSUE TITLE
ctrl+mouse wheel does not zoom in or out

### ISSUE THREAD

**Issue by safield** [freeplane/freeplane#134]
OS: Linux Mint 19.1
Freeplane: 1.7.7

I have just installed a fresh copy of freeplane 1.7.7.

When using freeplane on windows I can use ctrl+mousewheel to zoom in and out.

On my Linux Mint install, using ctrl+mousewheel, results in just panning up and down, as if I didn't click ctrl at all.

**Comment by jose1711** [comment]
tested with the current git (about says 1.7.8) on arch linux/gnome and it's working fine for me. you may want to try different display manager maybe (xfce)?

**Comment by safield** [comment]
Turns out it was result of using the application imwheel to modify the scroll amount per mouse wheel notch.

Unfortunately I deleted the exact imwheel commands I used when it broke freeplane (I forgot I even had this bug open).

It would be cool if they could work together though.

**Comment by soji256** [comment]
It appends the below to the imwheel config file (ex. "~/.imwheelrc").
```
Control_L, Up,   Control_L|Button4
Control_L, Down, Control_L|Button5
Control_R, Up,   Control_R|Button4
Control_R, Down, Control_R|Button5
```
That might work.

**Comment by Christopher-Hayes** [comment]
For anyone coming here after running `imwheel` standalone - creating `.imwheelrc` only with the config @soji256 provided doesn't work by itself. Add a line at the top with what applications that config should apply to, as seen in the Figma example from @emmanuelvlad. The config below worked for me:
```
"(google\-chrome)|(discord)|(brave)|(code)|(spotify)|(notion.*)|(mailspring)|(slack)|(inkscape)"
Control_L, Up,   Control_L|Button4
Control_L, Down, Control_L|Button5
Control_R, Up,   Control_R|Button4
Control_R, Down, Control_R|Button5
Shift_L,   Up,   Shift_L|Button4
Shift_L,   Down, Shift_L|Button5
```
Add any additional applications with `|(application-name)`. I also recommend adding configs for "Shift_L" as shown above for horizontal scrolling.

**Comment by pedrib** [comment]
Continuing on @soji256 and @Christopher-Hayes comment, if you want it to apply to all apps without having to specify them, just use:
```
".*"
Control_L, Up,   Control_L|Button4
Control_L, Down, Control_L|Button5
Control_R, Up,   Control_R|Button4
Control_R, Down, Control_R|Button5
```

**Comment by jorcelinojunior** [comment]
Hey there,

I ran into a similar issue on Ubuntu and found a workaround that might be helpful to others. After installing `imwheel` to fix a mouse scrolling issue, I noticed the scroll zoom and the forward/backward mouse buttons weren't working as expected.

Building on the solutions shared by @soji256 and @emmanuelvlad, I added a couple of lines to handle the forward and backward buttons on the mouse. Here's how you can do it:

1.  Open a terminal.
2.  Type `nano ~/.imwheelrc` to open the `.imwheelrc` file in a text editor.
3.  Replace the content of the file with the following configuration:

```bash
"(google-chrome)|(discord)|(brave)|(code)|(spotify)|(notion.*)|(mailspring)|(slack)|(inkscape)"
None,      Thumb1,   Alt_L|Left
None,      Thumb2,   Alt_L|Right
Control_L, Up,       Control_L|Button4
Control_L, Down,     Control_L|Button5
Control_R, Up,       Control_R|Button4
Control_R, Down,     Control_R|Button5
Shift_L,   Up,       Shift_L|Button4
Shift_L,   Down,     Shift_L|Button5
```

1.  Press `Ctrl+X` to close the editor, then press `Y` to save the changes.
2.  Restart `imwheel` with the command `killall imwheel && imwheel`.

This configuration maps `Thumb1` and `Thumb2` (usually the forward and backward buttons on the mouse) to the back and forward actions in most browsers and file managers.

To make sure `imwheel` starts on boot, you can add `imwheel` to your `.bashrc` file. Here's how:

1.  In the terminal, type `echo "imwheel" >> ~/.bashrc` or `echo "imwheel" >> ~/.zshrc` if you're using `zsh`.

I hope this helps anyone else running into the same issue!

References:
- [Edit your configuration file](https://wiki.archlinux.org/title/IMWheel#Edit_your_configuration_file)
- [How to solve a scrolling misbehavior after doing Alt+Tab?](https://askubuntu.com/a/1143466)
- [How do I start applications automatically on login?](https://askubuntu.com/a/48327)

**Comment by danieldownes** [comment]
Can anything be done to support this software, such displaying a warning message a dependency is needed, such as imwheel. 
Suggesting this as this post turns into Ubuntu support, rather than something to address in this software...

**Comment by dpolivaev** [comment]
I do not think that Freeplane should be aware of workarounds needed on different OS and different window managers.
People who discover such workarounds are encouraged to contribute pull requests to Freeplane online documentation. Probably this information could be added to page https://docs.freeplane.org/getting-started/Distributions.html?highlight=linux#distributions-for-linux. A button supporting it is available on each page there.
